sql/core/src/main/scala/org/apache/spark/sql/execution/joins/MergeAsOfJoinExec.scala

```diff
@@ -85,31 +92,38 @@ case class MergeAsOfJoinExec(
     var rPrev = rHead.copy()
 
     currLeft._2.map(lHead => {
```

Comment:
Using a mixture of map and while loop is not a very clean design. I think we should create sth similar to `SortMergeJoinScanner`

Reply:
You don't have to do it in this PR. Please refactor and submit another PR later.